File: common/src/main/java/com/hartwig/actin/treatment/input/single/ManyIntegers.java
package com.hartwig.actin.treatment.input.single;

import java.util.List;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class ManyIntegers {

    @NotNull
    public abstract List<Integer> integers();
}

File: common/src/main/java/com/hartwig/actin/treatment/input/single/ManyIntegers.kt
package com.hartwig.actin.treatment.input.single

import org.immutables.value.Value
import org.jetbrains.annotations.NotNull
import org.jetbrains.annotations.Nullable

@Value.Immutable
@Value.Style(passAnnotations = [NotNull::class, Nullable::class])
abstract class ManyIntegers() {
    abstract fun integers(): List<Int?>
}
